cypress/integration/7-crypto/crypto.spec.ts

```diff
@@ -19,12 +19,14 @@ limitations under the License.
 import type { MatrixClient } from "matrix-js-sdk/src/matrix";
 import { SynapseInstance } from "../../plugins/synapsedocker";
 
-function waitForEvent(cli: MatrixClient, event: any, check: () => boolean, callback: () => void) {
-    if (check()) {
-        callback();
-    } else {
-        cli.once(event, () => waitForEvent(cli, event, check, callback));
-    }
+function waitForEncryption(cli: MatrixClient, roomId: string, event: any, resolve: () => void) {
+    cli.crypto.cryptoStore.getEndToEndRooms(null, (result) => {
+        if (result[roomId]) {
+            resolve();
+        } else {
+            cli.once(event, () => waitForEncryption(cli, roomId, event, resolve));
+        }
+    });
```

Comment:
Shouldn't `event` be a string or something that extends a string rather than any?